File: JavaScript/8js 1/gnfs.js
/*
|--------------------------------------------------------------------------
| Get number from string
|--------------------------------------------------------------------------
|
*/

function getNumberFromString(s) {
   return Number(s.replace(/[^0-9]/g, ""));
}


function getNumberFromString(s) {
   return parseInt(
      s
         .split("")
         .filter((x) => "0123456789".includes(x))
         .join("")
   );
}


function getNumberFromString(s) {
   return +s.replace(/\D/g, "");
}


console.log(getNumberFromString("1"), 1);
console.log(getNumberFromString("123"), 123);
console.log(getNumberFromString("this is number: 7"), 7);

File: Python/8py 2/gnfs.py
"""
|--------------------------------------------------------------------------
| Get number from string
|--------------------------------------------------------------------------
|
"""

def get_number_from_string(string):
   return int(''.join([i for i in string if i.isdigit()]))


def get_number_from_string(s):
   return int(filter(str.isdigit, s))


from re import sub, findall
def get_number_from_string(s):
   return int(sub(r'\D', '', s))


def get_number_from_string(string):
   return int(''.join(findall(r'\d', string)))


print(get_number_from_string("1"), 1)
print(get_number_from_string("123"), 123)
print(get_number_from_string("this is number: 7"), 7)